conn.go

```diff
@@ -495,6 +518,16 @@ func (conn *Conn) sendReply(dest string, serial uint32, values ...interface{}) {
 	conn.outLck.RUnlock()
 }
 
+func (conn *Conn) defaultSignalAction(fn func(h *defaultSignalHandler, ch chan<- *Signal), ch chan<- *Signal) {
+	conn.signalLck.RLock()
```

Comment:
Can't this locking be done after the check for the default signal handler, or better yet put the lock inside the default signal handler?